lib/parse_date/int_from_string.rb

```diff
@@ -195,7 +209,7 @@ def year_from_mm_dd_yy(date_str)
       nil # explicitly want nil if date won't parse
     end
 
-    DECADE_4CHAR_REGEX = Regexp.new('(^|\D)\d{3}[u\-?x]', REGEX_OPTS)
+    DECADE_4CHAR_REGEX = Regexp.new('(^|\D)\d{3}[u\-?x]($|\D)', REGEX_OPTS)
```

Comment:
FYI - I don't grok these regexs (and am not putting in the effort to), but trust they work based on the test.